### DISCUSSION TITLE
How to listen on multi ports in one main loop? Is there a per-loop callback available?

### DISCUSSION THREAD

**Discussion by uniqss** [uNetworking/uWebSockets#970] [Q&A] [answered]
I am writing a gateway listening on multi ports for websocket connections, and a timer that must be called every loop. I don't want to use multi threads. Also I want to have a time manager that is not too precise which make a system call at the begining of every frame. The time manager also need to be called in the main loop. I just don't  know how to listen on multi ports and how to execute my per-frame callback. Thanks very much!

**Comment by ghost** [comment, marked as answer]
If you look at Loop.h there are addPostHandler etc. These run every new loop iteration.

One app can listen to many ports, you simply call listen many times

**Reply by uniqss** [reply]
Thanks very much~ It works.
